lib/tty/prompt/choice.rb

```diff
@@ -2,10 +2,12 @@
 
 module TTY
   class Prompt
-    # An immutable representation of a single choice option from select menu
+    # A representation of a single choice option from select menu
     #
     # @api public
     class Choice
+      attr_accessor :index
+
```

Comment:
I agree that adding index attribute and breaking immutability 'weakens' this class.